python/MDSplus/tree.py

```diff
@@ -1711,6 +1711,62 @@ def beginSegment(self, start, end, dim, array, idx=-1):
                                        _dat.Data.byref(array),
                                        _C.c_int32(int(idx))))
 
+    def beginSegmentResampled(self,start,end,dim,array,resNode,resFactor,idx=-1):
+        """Begin a resampled record segment, using an average, rows_filled = 0
+        @param start: Index of first row of data
+        @type start: Data
+        @param end: Index of last row of data
+        @type end: Data
+        @param dim: Dimension information of segment
+        @type dim: Dimension
+        @param array: Initial data array. Defines shape of segment
+        @type array: Array
+        @param resNode: TreeNode of the node to write the resampled data into
+        @type resNode: TreeNode
+        @param resFactor: Number of samples to resample together
+        @type resFactor: Data
+        @rtype: None
+        """
+        start,end,dim,array = map(_dat.Data,(start,end,dim,array))
+        _exc.checkStatus(
+            _TreeShr._TreeBeginSegmentResampled(self.ctx,
+                                                self._nid,
+                                                _dat.Data.byref(start),
+                                                _dat.Data.byref(end),
+                                                _dat.Data.byref(dim),
+                                                _dat.Data.byref(array),
+                                                _C.c_int32(int(idx)),
+                                                _C.c_int32(int(resNode._nid)),
```

Comment:
_nid is a ctypes pointer kind under python3 this wont work.. passing it directly like the second arguments should work though
```python
resNode._nid,
```